src/main/java/racing/Car.java

```diff
@@ -0,0 +1,58 @@
+package racing;
+
+public class Car implements Comparable<Car> {
+
+    private static final int NAME_LENGTH_CONDITION = 5;
+    private static final int MOVE_CONDITION = 4;
+
+    private final String name;
+    private int distance = 0;
+
+    public Car(String name) {
+        this.name = name;
+        validateNameLength(name);
```

Comment:
```suggestion
        this(name, distance)
        validateNameLength(name);
```
메소드 체이닝으로 수정해보는 것은 어떨까요?